src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpTagHelperFieldDeclarationVisitor.cs

```diff
@@ -47,28 +47,29 @@ protected override void Visit(TagHelperChunk chunk)
                     // Need to disable the warning "X is assigned to but never used." for the value buffer since
                     // whether it's used depends on how a TagHelper is used.
                     Writer.WritePragma("warning disable 0414");
-                    WritePrivateField(_tagHelperContext.TagHelperContentTypeName,
-                                      CSharpTagHelperCodeRenderer.StringValueBufferVariableName,
-                                      value: null);
+                    WritePrivateField(
+                        _tagHelperContext.TagHelperContentTypeName,
+                        CSharpTagHelperCodeRenderer.StringValueBufferVariableName,
+                        value: null);
                     Writer.WritePragma("warning restore 0414");
 
-                    WritePrivateField(_tagHelperContext.ExecutionContextTypeName,
-                                      CSharpTagHelperCodeRenderer.ExecutionContextVariableName,
-                                      value: null);
+                    WritePrivateField(
+                        _tagHelperContext.ExecutionContextTypeName,
+                        CSharpTagHelperCodeRenderer.ExecutionContextVariableName,
+                        value: null);
 
-                    Writer
-                        .Write("private ")
-                        .WriteVariableDeclaration(
+                    WritePrivateField(
                         _tagHelperContext.RunnerTypeName,
                         CSharpTagHelperCodeRenderer.RunnerVariableName,
                         value: null);
 
-                    Writer.Write("private ")
-                          .Write(_tagHelperContext.ScopeManagerTypeName)
-                          .Write(" ")
-                          .WriteStartAssignment(CSharpTagHelperCodeRenderer.ScopeManagerVariableName)
-                          .WriteStartNewObject(_tagHelperContext.ScopeManagerTypeName)
-                          .WriteEndMethodInvocation();
+                    Writer
+                        .Write("private global::")
+                        .Write(_tagHelperContext.ScopeManagerTypeName)
```

Comment:
It appears `RunnerTypeName` starts w/ "global::" but `ScopeManagerTypeName` does not. Instead of sometimes adding "global::" and sometimes not, change MVC to include the prefix in _all_ `_tagHelperContext.BlahTypeName` properties. Don't add the prefix here unless using `typeof()` or similar directly.